pandas/tests/indexes/datetimes/test_arithmetic.py

```diff
@@ -44,7 +44,23 @@ def addend(request):
 
 
 class TestDatetimeIndexComparisons(object):
-    # TODO: De-duplicate with test_comparisons_nat below
+    @pytest.mark.parametrize('other', [None,
+                                       datetime(2016, 1, 1).date()])
+    def test_dti_cmp_invalid(self, tz, other):
+        # GH#19301
+        dti = pd.date_range('2016-01-01', periods=2, tz=tz)
+
+        assert not (dti == other).any()
+        assert (dti != other).all()
+        with pytest.raises(TypeError):
+            dti < other
+        with pytest.raises(TypeError):
+            dti <= other
```

Comment:
this should also test ``pd.NaT`` here (and I think we have a tests for ``np.nan`` that should test raising).

Reply:
Just added nan to the params for this test.  pd.NaT test is immediately below this one.

Reply:
the datetime.date part of this needs to move to where we test comparisions with Timestamp, datetime.datetime and np.datetime64